gen/field.go

```diff
@@ -444,6 +450,17 @@ func (f fieldGroupGenerator) String(g Generator) error {
 		`, f)
 }
 
+func (f fieldGroupGenerator) ErrorName(g Generator) error {
+	return g.DeclareFromTemplate(
+		`
+		<$v := newVar "v">
+		// ErrorName returns a string representation of the type of <.Name>.
+		func (<$v> *<.Name>) ErrorName() string {
+			return "<.Name>"
```

Comment:
This will return the name of the Go struct. I think it'll be preferable here
to return the Thrift-level name in case it was renamed at the Go level,
especially if similar functionality gets added to other languages.

You'll need to plumb the Thrift-level name of the struct to the
fieldGroupGenerator.

Reply:
Added the tests and change the implementation to return the Thrift name instead.